enhancements/olm-deletes-operands.md

```diff
@@ -0,0 +1,193 @@
+---
+title: olm-deletes-operands
+authors:
+  - "@hasbro17"
+reviewers:
+  - "@benluddy"
+  - "@ecordell"
+  - "@kevinrizza"
+  - "@njhale"
+approvers:
+  - "@kevinrizza"
+  - "@dmesser"
+  - "@ecordell"
+  - "@shawn-hurley"  
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional
+see-also:
+  - "N/A"
+replaces:
+  - "N/A"
+superseded-by:
+  - "N/A"
+---
+
+# olm-deletes-operands
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The purpose of this enhancement is to outline how the [Operator Lifecycle Manager (OLM)][olm] can let operator users opt-in to the cleanup of their operator workloads, which would be cleaned up upon the uninstallation of their operator. Specifically when uninstalling an operator, OLM would delete all CustomResource (CR) objects managed by an operator so that the operator can be triggered to clean up all the resources associated with its CRs via [Finalizers][finalizers].
+
+## Motivation
+
+By design when OLM uninstalls an operator it does not remove any of the operator’s owned CustomResourceDefinitions (CRDs) and CRs in order to prevent data loss.
+The operator user or cluster admin is expected to manually delete CRs and any other secondary resources created by the operator.
+
+This cleanup can be incomplete when the operator is no longer available to process deletion events of CRs and handle cleanup via finalizers.
+Secondary resources that have their [OwnerReference][owner-reference] set to a CR would be garbage collected, but others such as off-cluster or cross-namespaced resources with no owner references would not be removed if the operator is already removed.
+
+So to facilitate automatically removing operator managed resources OLM can trigger cleanup via CR removal before the operator is removed.
+
+### Goals
+
+- Clean up includes removal of CRs (operands) and CRDs
+- CRs and CRDs are not automatically removed
+- A user can opt-in to the clean up behavior at operator uninstall time
+
+### Non-Goals
+
+- OLM handles cleanup of anything other than CRs
+- Handle custom cleanup logic like filtering/ordering the deletion of CRs
+
+### Possible Future Goals (Not in scope)
+
+- OLM should fire an alert event if the operator is stalled or taking too long on operand cleanup
+- The timeout for firing an alert event should be user configurable
```

Comment:
What's "user" here? The publisher of the operator or the OLM admin?

Reply:
Could be both actually. Although I was thinking of the operator user/author here who could set this on the CSV via `spec.cleanup` since they would be more familiar with the estimated cleanup time of their operands to set a reasonable alert timeout. 
And OLM can have a baked in default timeout in the absence of user/author configuration. 
But I'm punting on that issue to a follow up on this proposal.